designer/client/src/components/graph/node-modal/aggregate/aggMapLikeParser.tsx

```diff
@@ -1,236 +1,57 @@
-        const opening = this.CONSUME(LCurly);
-        const obj = this.SUBRULE(this.mapItems);
-        this.CONSUME(RCurly);
-
-        if (!opening.image || opening.isInsertedInRecovery) {
-            return null;
-        }
-
-        return obj;
-    });
-    private object = this.RULE("object", () => {
-        return this.OR([{ ALT: () => this.SUBRULE(this.aggMap) }, { ALT: () => this.SUBRULE(this.plainMap) }]);
-    });
-    private collection = this.RULE("collection", () => {
-        const arr = [];
-        this.CONSUME(CollectionOpen);
-        this.MANY_SEP({
-            SEP: Comma,
-            DEF: () => {
-                const item = this.SUBRULE(this.collectionItem);
-                if (!item) return;
-                arr.push(item);
-            },
-        });
-        this.CONSUME(CollectionClose);
-        return arr;
-    });
-    private list = this.RULE("list", () => {
-        const arr = [];
-        this.CONSUME(LCurly);
-        this.MANY_SEP({
-            SEP: Comma,
-            DEF: () => {
-                const item = this.SUBRULE(this.collectionItem);
-                if (!item) return;
-                arr.push(item);
-            },
-        });
-        this.CONSUME(ListClose);
-        return arr;
-    });
-    private groupBy = this.RULE("groupBy", () => {
-        return this.OR([{ ALT: () => this.SUBRULE(this.collection) }, { ALT: () => this.SUBRULE(this.list) }]);
-    });
-    private fullText = "";
-
-    constructor(tokens = aggMapTokens) {
-        super(tokens, { recoveryEnabled: true });
-        this.lexer = new Lexer(tokens);
-        this.parseObject = withLogs(this.parseObject.bind(this));
-        this.parseList = withLogs(this.parseList.bind(this));
-        this.performSelfAnalysis();
     }
 
-    parseList(input: string): Array<string> {
-        this.tokenizeInput(input);
-        return this.groupBy() || null;
+    #parseList(input: string, ast?: Ast) {
+        if (ast?.type !== "CompoundExpression") return null;
+        if (ast.expressionComponents[0].type !== "InlineList") return null;
+        if (ast.expressionComponents[1].type !== "PropertyReference") return null;
+        if (ast.expressionComponents[1].propertyName !== "toString") return null;
+        return ast.expressionComponents[0].elements.map((el) => this.#printFragment(input, el));
     }
 
-    parseObject(input: string): Record<string, string> {
-        this.tokenizeInput(input);
-        return this.object() || null;
+    #printFragment(
+        text: string,
+        el: {
+            start: number;
+            end: number;
+        },
+    ) {
+        return text.slice(el.start, el.end).trim();
     }
 
-    private tokenizeInput(input: string) {
-        const lexResult = this.lexer.tokenize(input);
-        this.fullText = input;
-        this.input = lexResult.tokens;
+    #parseObject(input: string, ast?: Ast) {
+        switch (ast?.type) {
+            case "InlineList":
```

Comment:
Let's write a note that it is a work around for a bug in the ts-spel parser that it treats empty map as empty list (I'm not sure If I understand you correctly)

Reply:
it's not a bug. it's the way it works. {} means empty list. i want to treat empty list as empty object in this case - that's all. there is no sense and case where ":" would be inserted. test case shows this clearly.